blockchain/tx_pool.go

```diff
@@ -779,9 +783,26 @@ func (pool *TxPool) validateTx(tx *types.Transaction) error {
 		return ErrNonceTooLow
 	}
 
-	// Transactor should have enough funds to cover the costs
-	// cost == V + GP * GL
+	// balance check for module transaction if module have check balance function
+	checkedBalance := false
+	for _, module := range pool.modules {
+		if module.IsModuleTx(pool, tx) {
+			if checkBalance := module.GetCheckBalance(); checkBalance != nil {
+				checkedBalance = true
+				err := checkBalance(pool, tx)
+				if err != nil {
+					logger.Trace("[tx_pool] invalid funds of module transaction sender", "from", from, "txhash", tx.Hash().Hex())
+					return err
```

Comment:
If a module returns checkBalance function, it will override the default balance check (by setting `checkedBalance=true`).
What if a module doesn't want to ignore the default balance check?

Reply:
You mean the situation where a module want additional check in addition to the default logic?

Reply:
Right. For now, we can leave it as-is, but let's consider it later

Reply:
Yes. Let's consider things like this _later_.
```
module.ShouldOverrideBalanceCheck(tx) -> bool
module.BalanceCheck(tx)
```